src/Storefront/Framework/Twig/TwigAppVariable.php

```diff
@@ -21,6 +21,8 @@ class TwigAppVariable extends AppVariable
 
     private array $allowList;
 
+    private array $flashes = [];
```

Comment:
This needs to be cleared after the request. Use the `kernel.reset` tag + ResetInterface to make it sure

Reply:
tbh: this really looks like a really wrong solution. We should not have somewhere a state of flash messages.

You should rather assign the flashes to a variable inside the twig template. 

I could also think that we can add the flashes to the page object in `renderStorefront`. But applying more custom logic to symfony default seems not so good here

Reply:
Yeah, it feels whacky but so far my best guess. I have similar thoughts. That is why I added the hint in my description, that it is a bit off of Symfony.

I don't think we should add the flashes to the page object. This will empty the flashbag although we don't know whether they will be rendered.

We already have a flashes variable in twig which is called messages. It is pulled at the point where some flashes are displayed. But this is quite late in the overall rendering. It should be more prominent in the first few lines of twig, but this eventually runs us into the same issue: no one can ensure they will be rendered.

So eventually we just should stay with the approach to pull the flashes out of the bag, when they are really displayed. Otherwise they get lost. Like the "product not found" error in https://github.com/shopware/platform/pull/2693 

Any other ideas?

Reply:
We will go with the TwigAppVariable way and implement resetinterface to make sure it's empty when we ever try to use roadrunner.

Also provide a PR to Symfony directly as this should be in the framework

Reply:
https://github.com/symfony/symfony/pull/49042

Rebased and added reset interface :)